src/hal-api-ml.c

```diff
@@ -180,6 +189,15 @@ hal_ml_create (const char *backend_name, hal_ml_h *handle)
   static int scanned = 1;
   if (scanned == 1) {
     scanned = hal_ml_scan_backends ();
+    if (scanned < 0) {
+      _E ("Failed to scan backends");
+      return HAL_ML_ERROR_RUNTIME_ERROR;
+    }
+  }
+
+  if (hal_ml_backend_count == 0) {
```

Comment:
scanned?
```suggestion
  if (scanned== 0) {
```

Reply:
`scanned == 0` means the scanning process is done before.